extend/component/implementation/python.md

```diff
@@ -111,82 +157,254 @@ with open('in/tables/source.csv', mode='rt', encoding='utf-8') as in_file, open(
-        pk = ['primaryKey']
-    else:
-        pk = manifest['primary_key']
-
-    # write table metadata - set new primary key
-    cfg.write_table_manifest(outName, destination=outDestination, primary_key=pk)
-    j = j + 1
-{% endhighlight %}
+    outDestination = ci.configuration.tables_output_mapping[j]['destination']
+```
+
+### I/O table manifests and processing results
+
+The component may define output [manifest files](https://developers.keboola.com/extend/common-interface/manifest-files/#dataouttables-manifests) 
+that define options on storing the results back to the Keboola Connection Storage. This library provides methods that simplifies 
+the manifest file creation and allows defining the export options and metadata of the result table using helper objects `TableDefinition` 
+and `TableMetadata`.
+
+`TableDefinition` object serves as a result container containing all the information needed to store the Table into the Storage. 
+It contains the manifest file representation and initializes all attributes available in the manifest.
+This object represents both Input and Output manifests. All output manifest attributes are exposed in the class.
+
+There are convenience method for manifest creation `CommonInterface.write_tabledef_manifest()`. 
+Also it is possible to create the container for the output table using the `CommonInterface.create_out_table_definition()`
+(useful particularly when working with [sliced tables](/extend/common-interface/folders/#sliced-tables)).
+
+```python
+from keboola.component import CommonInterface
+from keboola.component import dao
+
+# init the interface
+ci = CommonInterface()
+
+# create container for the result
+result_table = ci.create_out_table_definition('my_new_result_table', primary_key=['id'], incremental=True)
+
+# write some content
+with open(result_table.full_path, 'w') as result:
+    result.write('line')
+
+# add some metadata
+result_table.table_metadata.add_table_description('My new table description')
+# add column datatype
+result_table.table_metadata.add_column_data_type('id', dao.SupportedDataTypes.STRING, 
+                                                 source_data_type='VARCHAR(100)', 
+                                                 nullable=True,
+                                                 length=100)
+
+# write manifest
+ci.write_tabledef_manifest(result_table)
+```
+
+#### Get input table by name
+
+```python
+from keboola.component import CommonInterface
+
+# init the interface
+ci = CommonInterface()
+table_def = ci.get_input_table_definition_by_name('input.csv')
+```
+
+#### Initializing TableDefinition object from the manifest file
+
+```python
+from keboola.component import dao
+
+table_def = dao.TableDefinition.build_from_manifest('data/in/tables/table.csv.manifest')
+
+# print table.csv full-path if present:
+print(table_def.full_path)
+
+# rows count
+print(table_def.rows_count)
+```
+
+#### Retrieve raw manifest file definition (CommonInterface compatible)
```

Comment:
I don't understand what this is for

Reply:
Hi,  I'm terribly sorry for letting it rot here for so long:( This `get_manifest_dictionary()` is being used to get the Keboola manifest file representation of the TableDefinition object. Some of the properties are kept in different structures that are more convenient to work with (TableMetadata) so this will convert it to the proper manifest file dictionary that can be stored as JSON. It's being used in [write_tabledef_manifest()](https://github.com/keboola/python-component/blob/main/src/keboola/component/interface.py#L425) method for instance. 

Now I see it's confusing. Maybe we should exclude this all together along with the previous example of `build_from_manifest`. This is being used internally by `get_input_tables_definitions()`/'get_input_files_definitions' and normal user would use only these methods to load the input tables along with anything that's in the manifest.

Reply:
I removed both sections in 9cd3e60